### ISSUE TITLE
IDE error when switching to Default JavaFX Stylesheet

### ISSUE THREAD

**Issue by gkertai** [vsch/idea-multimarkdown#348]
Version 2.2.0

In *Preferences / Languages & Frameworks / Markdown / Stylesheet* I changed the *Stylesheet* setting from **Default Swing Stylesheet** to **Default JavaFX Stylesheet**, and an IDE error was reported when I clicked Apply. Also, the markdown preview no longer rendered.

```
Unclosed comment
java.lang.RuntimeException: Unclosed comment
	at javax.swing.text.html.CSSParser.readComment(CSSParser.java:743)
	at javax.swing.text.html.CSSParser.getIdentifier(CSSParser.java:609)
	at javax.swing.text.html.CSSParser.nextToken(CSSParser.java:505)
	at javax.swing.text.html.CSSParser.getNextStatement(CSSParser.java:168)
	at javax.swing.text.html.CSSParser.parse(CSSParser.java:153)
	at javax.swing.text.html.StyleSheet$CssParser.parse(StyleSheet.java:3187)
	at javax.swing.text.html.StyleSheet.addRule(StyleSheet.java:307)
	at javax.swing.text.html.HTMLDocument$HTMLReader.addCSSRules(HTMLDocument.java:3828)
	at javax.swing.text.html.HTMLDocument$HTMLReader$HeadAction.end(HTMLDocument.java:2965)
	at javax.swing.text.html.HTMLDocument$HTMLReader.handleEndTag(HTMLDocument.java:2703)
	at javax.swing.text.html.parser.DocumentParser.handleEndTag(DocumentParser.java:240)
	at javax.swing.text.html.parser.Parser.parse(Parser.java:2290)
	at javax.swing.text.html.parser.DocumentParser.parse(DocumentParser.java:122)
	at javax.swing.text.html.parser.ParserDelegator.parse(ParserDelegator.java:102)
	at javax.swing.text.html.HTMLEditorKit.read(HTMLEditorKit.java:261)
	at javax.swing.JEditorPane.setText(JEditorPane.java:1416)
	at com.vladsch.idea.multimarkdown.editor.swing.SwingHtmlPanel.setHtml(SwingHtmlPanel.kt:193)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase$updateHtmlRunner$8.run(PreviewFileEditorBase.kt:441)
	at com.vladsch.idea.multimarkdown.util.TimeIt.logTime(TimeIt.java:28)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase.updateHtmlRunner(PreviewFileEditorBase.kt:440)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase.access$updateHtmlRunner(PreviewFileEditorBase.kt:60)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase$updateHtml$$inlined$synchronized$lambda$1.run(PreviewFileEditorBase.kt:361)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:223)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:387)
	at com.intellij.openapi.application.TransactionGuardImpl$6.run(TransactionGuardImpl.java:318)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:417)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:401)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


java.lang.RuntimeException: Unclosed comment
	at javax.swing.text.html.CSSParser.readComment(CSSParser.java:743)
	at javax.swing.text.html.CSSParser.getIdentifier(CSSParser.java:609)
	at javax.swing.text.html.CSSParser.nextToken(CSSParser.java:505)
	at javax.swing.text.html.CSSParser.getNextStatement(CSSParser.java:168)
	at javax.swing.text.html.CSSParser.parse(CSSParser.java:153)
	at javax.swing.text.html.StyleSheet$CssParser.parse(StyleSheet.java:3187)
	at javax.swing.text.html.StyleSheet.addRule(StyleSheet.java:307)
	at javax.swing.text.html.HTMLDocument$HTMLReader.addCSSRules(HTMLDocument.java:3828)
	at javax.swing.text.html.HTMLDocument$HTMLReader$HeadAction.end(HTMLDocument.java:2965)
	at javax.swing.text.html.HTMLDocument$HTMLReader.handleEndTag(HTMLDocument.java:2703)
	at javax.swing.text.html.parser.DocumentParser.handleEndTag(DocumentParser.java:240)
	at javax.swing.text.html.parser.Parser.parse(Parser.java:2290)
	at javax.swing.text.html.parser.DocumentParser.parse(DocumentParser.java:122)
	at javax.swing.text.html.parser.ParserDelegator.parse(ParserDelegator.java:102)
	at javax.swing.text.html.HTMLEditorKit.read(HTMLEditorKit.java:261)
	at javax.swing.JEditorPane.setText(JEditorPane.java:1416)
	at com.vladsch.idea.multimarkdown.editor.swing.SwingHtmlPanel.setHtml(SwingHtmlPanel.kt:193)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase$updateHtmlRunner$8.run(PreviewFileEditorBase.kt:441)
	at com.vladsch.idea.multimarkdown.util.TimeIt.logTime(TimeIt.java:28)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase.updateHtmlRunner(PreviewFileEditorBase.kt:440)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase.access$updateHtmlRunner(PreviewFileEditorBase.kt:60)
	at com.vladsch.idea.multimarkdown.editor.PreviewFileEditorBase$updateHtml$$inlined$synchronized$lambda$1.run(PreviewFileEditorBase.kt:361)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:223)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:387)
	at com.intellij.openapi.application.TransactionGuardImpl$6.run(TransactionGuardImpl.java:318)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:417)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:401)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

Switching back to *Default Swing Stylesheet* or *No Stylesheet* resolves the issue. Reproducible 100% of the time. I am using the JavaFX rendering engine.

**Comment by vsch** [comment]
@gkertai, you are using swing browser. I will check why it shows JavaFX in settings because the code is failing for JavaFX and switching to Swing, but not switching the style sheet. That is the cause of the error. It is all Swing HTMLEditorKit code:

```
java.lang.RuntimeException: Unclosed comment
	at javax.swing.text.html.CSSParser.readComment(CSSParser.java:743)
	at javax.swing.text.html.CSSParser.getIdentifier(CSSParser.java:609)
	at javax.swing.text.html.CSSParser.nextToken(CSSParser.java:505)
	at javax.swing.text.html.CSSParser.getNextStatement(CSSParser.java:168)
	at javax.swing.text.html.CSSParser.parse(CSSParser.java:153)
	at javax.swing.text.html.StyleSheet$CssParser.parse(StyleSheet.java:3187)
	at javax.swing.text.html.StyleSheet.addRule(StyleSheet.java:307)
	at javax.swing.text.html.HTMLDocument$HTMLReader.addCSSRules(HTMLDocument.java:3828)
	at javax.swing.text.html.HTMLDocument$HTMLReader$HeadAction.end(HTMLDocument.java:2965)
	at javax.swing.text.html.HTMLDocument$HTMLReader.handleEndTag(HTMLDocument.java:2703)
	at javax.swing.text.html.parser.DocumentParser.handleEndTag(DocumentParser.java:240)
	at javax.swing.text.html.parser.Parser.parse(Parser.java:2290)
	at javax.swing.text.html.parser.DocumentParser.parse(DocumentParser.java:122)
	at javax.swing.text.html.parser.ParserDelegator.parse(ParserDelegator.java:102)
	at javax.swing.text.html.HTMLEditorKit.read(HTMLEditorKit.java:261)
	at javax.swing.JEditorPane.setText(JEditorPane.java:1416)
```

**Comment by gkertai** [comment]
You are right. Somehow it switched back to Swing… Maybe after the latest PHPStorm upgrade? Switching to JavaFX resolves the problem.

**Comment by vsch** [comment]
@gkertai, yes I noticed loosing a few settings here and there after an IDE upgrade. I have had to manually delete the IDE cache directories in: `~/Library/Application Support`, `~/Library/Caches` and `~/Library/Preferences` a few times to let it start with a clean slate or it would not work reliably, like fail compilation of project with cryptic error message about a missing library that is not used in the project.

Even the IDE I build from open source intellij-community, somehow, after the last source update and build, managed to loose workspace information for my projects. I open them up and I get funny sized windows, no recent files and keymap reset to default. It was not even a version upgrade just a dozen update files and project build.
